### PR TITLE
feat: modify gha to test and publish alpine based images

### PR DESCRIPTION
**Pull request by alles-klar**
We already have the alpine based django image. The PR adopts the pipeline and allows to push the new images to dockerhub. The debian image is still the default image with the latest tag.

We will have three tags in dockerhub:
- latest (at the moment, the (current) debian based images)
- [defectdojo_version]-debian (current)
- [defectdojo_version]-alpine (new)

Disadvantage: The nginx image which was already based on alpine is now called nginx-debian. This was required to distinguish between the old nginx image, where the build process was based on debian, and the new image, where also the build process is based on alpine.

Please review carefully, if I haven't broken our pipeline!

---
On behalf of DB Systel GmbH

### CODE REVIEW

**Comment by Maffooch** [comment]
@alles-klar I think the approach for separating the names of the docker files into alpine and debian makes sense, but I believe it will bring more pain that benefit. 

Every user will be forced to upgrade their compose and make a choice for debian/alpine. I think it would be simpler to keep the debian images as Django/nginx and the alpine images to be labeled as Django-alpine/nginx-alpine. A note in the upgrade docs would be useful to inform users they have the choice to change it if they desire.

**Comment by alles-klar** [comment]
@Maffooch you are right I forgot to update the dockerfile names in the compose files. I have done that and also adding some lines to the docs.

The old docker tags are still the same, so hopefully there are no changes for the standard way of using defectdojo.

**Comment by alles-klar** [comment]
@Maffooch there will be some more, you can see them in line 134 and 149 of release-2-tag-docker-push.yml.

Dockerfile.django-debian:
* defectdojo/defectdojo-django:latest
* defectdojo/defectdojo-django:2.19.0
* defectdojo/defectdojo-django:2.19.0-debian

Dockerfile.nginx-debian:
* defectdojo/defectdojo-nginx:latest
* defectdojo/defectdojo-nginx:2.19.0
* defectdojo/defectdojo-nginx:2.19.0-debian

Dockerfile.django-alpine:
* defectdojo/defectdojo-django:2.19.0-alpine

Dockerfile.nginx-alpine:
* defectdojo/defectdojo-nginx:2.19.0-alpine

**Comment by Maffooch** [comment]
That's works perfectly!

**Comment by gietschess** [comment]
Looks good! And makes #7222 needlees as we can change the distribution with the tag helm value directly.

File: docs/content/en/getting_started/upgrading.md [hunk 1]
```diff
@@ -61,6 +68,10 @@ godojo installations
 
 If you have installed DefectDojo on "iron" and wish to upgrade the installation, please see the [instructions in the repo](https://github.com/DefectDojo/godojo/blob/master/docs-and-scripts/upgrading.md).
 
+## Upgrading to DefectDojo Version 2.19.x
+
+There are new dockerimages based on alpine with less vulnerabilities. Related to the new images the current dockerfiles had to be renamed and have a "-debian" or the new images a "-alpine" at the end. Furthermore there are new docker tags [DefectdojoVersion]-[OS]. For example 2.19.0-alpine or 2.19.0-debian. The currend tags (latest and [DefectdojoVersion]) are still based on the "old" images. Be aware that the new alpine images are not heavily tested and may contain bugs.
```
Comment (Maffooch) [inline docs/content/en/getting_started/upgrading.md:73]:
```suggestion
There are new docker images based on alpine with fewer third party dependencies. Related to the new images the current docker files had to be renamed and have a "-debian" or the new images a "-alpine" at the end. Furthermore there are new docker tags [DefectdojoVersion]-[OS]. For example 2.19.0-alpine or 2.19.0-debian. The currend tags (latest and [DefectdojoVersion]) are still based on the "old" images. Be aware that the new alpine images are not heavily tested and may contain bugs.
```

**Comment by mtesauro** [comment]
This LGTM.  Dropping my approval  now.

@dsever Thoughts?

**Review by mtesauro — APPROVED** [review]
Approved

File: Dockerfile.nginx-alpine [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+COPY wsgi_params nginx/nginx.conf nginx/nginx_TLS.conf /etc/nginx/
+COPY docker/entrypoint-nginx.sh /
+RUN \
+  apk add --no-cache openssl && \
+  chmod -R g=u /var/cache/nginx && \
+  mkdir /var/run/defectdojo && \
+  chmod -R g=u /var/run/defectdojo && \
+  mkdir -p /etc/nginx/ssl && \
+  chmod -R g=u /etc/nginx && \
+  true
```
Comment (dsever) [inline Dockerfile.nginx-alpine:156]: Why true at the end?  
```
false && true  && true;  echo $?
1
```

File: docs/content/en/getting_started/upgrading.md [hunk 3]
```diff
@@ -37,6 +37,13 @@ The generic upgrade method for docker-compose follows these steps:
     docker pull defectdojo/defectdojo-nginx:1.10.2
     ```
 
+-   If you would like to use alpine based images, you specify the version (tag) you want to upgrade to:
+
+    ``` {.sourceCode .bash}
+    docker pull defectdojo/defectdojo-django:1.10.2-alpine
+    docker pull defectdojo/defectdojo-nginx:1.10.2-alpine
+    ```
```
Comment (dsever) [inline docs/content/en/getting_started/upgrading.md:45]: I like it

File: Dockerfile.nginx-alpine (hunk 2 above)
Comment (alles-klar) [reply Dockerfile.nginx-alpine:156]: It is just copy and past from our current nginx image. If we want to change that, maybe not in this PR to keep it simple.

File: Dockerfile.nginx-alpine [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    && adduser -u 1000 -G node -s /bin/sh -D node \
+    && apk add --no-cache \
+        libstdc++ \
+    && apk add --no-cache --virtual .build-deps \
+        curl \
+    && ARCH= && alpineArch="$(apk --print-arch)" \
+      && case "${alpineArch##*-}" in \
+        x86_64) \
+          ARCH='x64' \
+          CHECKSUM="069d9b0f9c01fa269dfe71e0b65f04a151c233c81bd998b59894c5f717f79877" \
```
[the remark this answers is not in the document]
Comment (alles-klar) [reply Dockerfile.nginx-alpine:49]: It is just a simple copy from the docker-node project: https://github.com/nodejs/docker-node/blob/main/14/alpine3.16/Dockerfile

I thought relying on proved code is better and easier than reinventing the wheel. But of course we can build our own installation code. But than we have to change more than this single line. 

For the moment I would prefer to leave it as it is.

[the remark this answers is not in the document]
Comment (dsever) [reply Dockerfile.nginx-alpine:49]: I'm afraid that it will be nightmare to maintenance the changes

[the remark this answers is not in the document]
Comment (dsever) [reply Dockerfile.nginx-alpine:49]: Please take a look at the https://github.com/nodejs/docker-node/blob/main/Dockerfile-alpine.template  and this https://github.com/nodejs/docker-node/blob/main/update.sh even they are not hard-coding, they are using templates.

[the remark this answers is not in the document]
Comment (alles-klar) [reply Dockerfile.nginx-alpine]: Good point, seems I missed that.

Creating a template is maybe not the best option for us, but I changed it to a dynamic checksum check.

**Review by dsever — APPROVED** [review]
Let's go with it

**Pull request merged**